src/main/java/eu/europeana/metis/sandbox/service/workflow/MediaProcessingServiceImpl.java

```diff
@@ -76,21 +82,45 @@ public RecordInfo processMedia(Record record) {
     return new RecordInfo(Record.from(record, outputRdf), recordErrors);
   }
 
-  private void processResourceEntry(Record record, EnrichedRdf rdfForEnrichment,
-      MediaExtractor extractor,
-      RdfResourceEntry entry, List<RecordError> recordErrors) {
-    try (ResourceExtractionResult extraction = extractor.performMediaExtraction(entry)) {
-      if (nonNull(extraction)) {
-        // Store thumbnails
-        storeThumbnails(record, extraction.getThumbnails(), recordErrors);
-        // Add result to RDF
+  /**
+   * Processes the resource
+   *
+   * @return True if the media extraction for the resource was successful; False otherwise.
+   */
+  private boolean processResource(RdfResourceEntry resourceToProcess, Record record,
+      EnrichedRdf rdfForEnrichment, MediaExtractor extractor, List<RecordError> recordErrors,
+      boolean hasRanMainThumbnail, boolean gotMainThumbnail){
+
+    ResourceExtractionResult extraction;
+    boolean successful = false;
+
+    try{
+
+      // Check if the process for main thumbnail has happened.
+      // If not, then get main thumbnail
+      if(!hasRanMainThumbnail) {
+        extraction = extractor.performMediaExtraction(resourceToProcess, false);
+      } else {
+        extraction = extractor.performMediaExtraction(resourceToProcess, gotMainThumbnail);
+      }
+
+      // Check if extraction for media was successful
+      successful = extraction != null && !extraction.getThumbnails().isEmpty();;
```

Comment:
Be careful here: success is not defined by whether or not there are thumbnails! Some resources never have thumbnails (such as audio files). You will need an extra if statement within the if below to check for the availability of thumbnails.

Reply:
Good point, I will fix this